File: docker/Dockerfile.java
# Dockerfile for testing Java SDK examples
FROM maven:3.9-eclipse-temurin-17

WORKDIR /app

# Copy Java SDK
COPY sdks/java /app/sdks/java

# Build SDK
WORKDIR /app/sdks/java
RUN mvn clean package -DskipTests

# Run examples
CMD ["sh", "-c", "mvn exec:java -Dexec.mainClass=dev.windjammer.examples.HelloWorld && mvn exec:java -Dexec.mainClass=dev.windjammer.examples.SpriteDemo && mvn exec:java -Dexec.mainClass=dev.windjammer.examples.Scene3D"]

File: docker/Dockerfile.cpp
# Dockerfile for testing C++ SDK examples
FROM gcc:16

WORKDIR /app

# Install CMake
RUN apt-get update && apt-get install -y \
    cmake \
    ninja-build \
    git \
    && rm -rf /var/lib/apt/lists/*

# Copy C++ SDK
COPY sdks/cpp /app/sdks/cpp

# Build examples
WORKDIR /app/sdks/cpp
RUN mkdir -p build && cd build && \
    cmake .. -GNinja && \
    ninja

# Run tests
CMD ["sh", "-c", "./build/examples/hello_world && ./build/examples/math_demo && ./build/examples/sprite_demo && ./build/examples/3d_scene"]
